docs/Node/ReadLine.md

```diff
@@ -12,37 +12,85 @@ A handle to a console interface.
 
 A handle can be created with the `createInterface` function.
 
+#### `READLINE`
+
+``` purescript
+data READLINE :: !
+```
+
+The effect of interacting with a stream via an `Interface`
+
+#### `InterfaceOptions`
+
+``` purescript
+data InterfaceOptions
+```
+
+Options passed to `readline`'s `createInterface`
+
+#### `output`
+
+``` purescript
+output :: forall w eff. Option InterfaceOptions (Writable w eff)
+```
+
+#### `completer`
+
+``` purescript
+completer :: forall eff. Option InterfaceOptions (Completer eff)
+```
+
+#### `terminal`
+
+``` purescript
+terminal :: Option InterfaceOptions Boolean
+```
+
+#### `historySize`
+
+``` purescript
+historySize :: Option InterfaceOptions Int
+```
+
 #### `Completer`
 
 ``` purescript
-type Completer eff = String -> Eff (console :: CONSOLE | eff) { completions :: Array String, matched :: String }
+type Completer eff = String -> Eff eff { completions :: Array String, matched :: String }
 ```
 
 A function which performs tab completion.
 
-This function takes the partial command as input, and returns a collection of 
+This function takes the partial command as input, and returns a collection of
 completions, as well as the matched portion of the input string.
 
-#### `LineHandler`
+#### `createInterface`
 
 ``` purescript
-type LineHandler eff a = String -> Eff (console :: CONSOLE | eff) a
+createInterface :: forall r eff. Readable r eff -> Options InterfaceOptions -> Eff (readline :: READLINE | eff) Interface
```

Comment:
This has languished a bit, sorry about that. I think we can safely merge this now and address all my other comments later, but this one probably does want attention. If you change the first argument to `Readable r (readline :: READLINE | eff)`, then I'll merge.